lib/src/models/gateway/events/invite.dart

```diff
@@ -27,5 +29,11 @@ class InviteDeleteEvent extends DispatchEvent {
   final String code;
 
   /// {@macro invite_delete_event}
-  InviteDeleteEvent({required this.channelId, required this.guildId, required this.code});
+  InviteDeleteEvent({required super.gateway, required this.channelId, required this.guildId, required this.code});
+
+  /// The channel the invite was for.
+  PartialChannel get channel => gateway.client.channels[channelId];
```

Comment:
you passed to every event gatewa, but is there a need for gateway? Could it be just client?

Reply:
The event objects don't have a reference to the client, only a reference to the gateway that than can reference the client. We could add a client getter but I don't think we need to shorten gateway.client to just client.

Reply:
I was thinking about just passing client ot the event because I presume event doesnt need gateway specifically and its there just to get client. It would make leaner dependencies and easier mocking/extending in future if needed

Reply:
It's more consistent to pass gateway because Gateway is essentially the manager for all events - for all other structures we pass the manager so this is "the same" here.

Reply:
aight you got me